### PR TITLE
Fix compile error - no member named 'binary_negate' in namespace 'std'

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we fix errors when in C++20 (Windows, Intel(R) oneAPI DPC++/C++ Compiler 2022.0.0 (2022.0.0.20211123) ) in file [include/oneapi/dpl/functional](https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/functional#L51) we include some functions,
which are absent in Standard library:
```
In file included from test\parallel_api\algorithm\alg.modifying.operations\rotate.pass.cpp:19:
In file included from include\oneapi/dpl/algorithm:37:
In file included from include\oneapi/dpl/internal/exclusive_scan_by_segment_impl.h:23:
In file included from include\oneapi/dpl/internal/function.h:23: include\oneapi/dpl/internal/../functional(51,14): error: no member named 'binary_negate' in namespace 'std'
       using ::std::binary_negate; // Deprecated in C++17, removed in C++20
       ~~~~~~~^
```

Which functions absent in Standard library in this case:
```
#if _ONEDPL___cplusplus < 202002L
using ::std::binary_negate; // Deprecated in C++17, removed in C++20
using ::std::not1;          // Deprecated in C++17, removed in C++20
using ::std::not2;          // Deprecated in C++17, removed in C++20
using ::std::unary_negate;  // Deprecated in C++17, removed in C++20
#endif                      // _ONEDPL___cplusplus < 202002L
```

Why this happened: because In problem configuration with Intel(R) oneAPI DPC++/C++ Compiler 2022.0.0 (2022.0.0.20211123)
we have the next state of defines (with command line options /Zc:__cplusplus  -std:c++latest):
```
_MSVC_LANG: 201705
__cplusplus: 202101
```
Our previous implementation not correspondent to this state:
```
#ifndef _MSVC_LANG
#    define _ONEDPL___cplusplus __cplusplus
#else
#    define _ONEDPL___cplusplus _MSVC_LANG
#endif
```

This issue was introduced in PR [#678](https://github.com/oneapi-src/oneDPL/pull/678)

### CODE REVIEW

**Review by MikeDvorskiy — COMMENTED** [review]
Looks reasonable for me.
But  lets one more reviewer has a look...

File: include/oneapi/dpl/internal/common_config.h [hunk 1]
```diff
@@ -19,7 +19,11 @@
 #ifndef _MSVC_LANG
 #    define _ONEDPL___cplusplus __cplusplus
 #else
-#    define _ONEDPL___cplusplus _MSVC_LANG
+#    if _MSVC_LANG > __cplusplus
+#        define _ONEDPL___cplusplus _MSVC_LANG
+#    else
+#        define _ONEDPL___cplusplus __cplusplus
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/internal/common_config.h:25]: Please see details in [](https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170#standard-predefined-macros) :

File: test/support/test_config.h [hunk 2]
```diff
@@ -106,7 +100,7 @@
 #define _PSTL_GLIBCXX_TEST_COMPLEX_PLUS_EQ_BROKEN _PSTL_GLIBCXX_TEST_COMPLEX_BROKEN
 #define _PSTL_GLIBCXX_TEST_COMPLEX_TIMES_EQ_BROKEN _PSTL_GLIBCXX_TEST_COMPLEX_BROKEN
 
-#define _PSTL_MSVC_LESS_THAN_CPP20_COMPLEX_CONSTEXPR_BROKEN (_MSC_VER && _ONEDPL_TEST___cplusplus <= 201703L)
+#define _PSTL_MSVC_LESS_THAN_CPP20_COMPLEX_CONSTEXPR_BROKEN (_MSC_VER && !(__cplusplus > 201703L || _MSVC_LANG > 201703L))
```
Comment (rarutyun) [inline test/support/test_config.h:103]: It's better to make < 202002L (or whatever it is) check. Probably even better to use `TEST_VER` macro for that.

Comment (SergeyKopienko) [reply test/support/test_config.h]: Fixed.

**Review by rarutyun — APPROVED** [review]

**Pull request merged**